### PR TITLE
[REPO-5552] more filtering capablities for aspect/type api

### PR DESCRIPTION
**Pull request by dhrn**
depends https://github.com/Alfresco/rest-api-explorer/pull/125

### CODE REVIEW

File: remote-api/src/main/java/org/alfresco/rest/api/impl/AspectsImpl.java [hunk 1]
```diff
@@ -125,32 +138,50 @@ public Aspect getAspectById(String aspectId)
...
-        return new Aspect(aspectDefinition, dictionaryService, properties);
+        try
+        {
+            Aspect aspect = new Aspect(aspectDefinition, dictionaryService);
+            constructFromFilters(aspect, aspectDefinition, includes);
+            return aspect;
+        }
+        catch (Exception ex)
+        {
+            logger.warn("Failed to parse Aspect " + aspectDefinition.getName().toPrefixString());
```
Comment (gavincornwell) [inline remote-api/src/main/java/org/alfresco/rest/api/impl/AspectsImpl.java:154]: Swallowing errors at the API level is generally not a good idea, the user needs to know they made a mistake. What causes this error? It should be caught and re-thrown so the appropriate HTTP error code is returned.

File: remote-api/src/main/java/org/alfresco/rest/api/impl/TypesImpl.java [hunk 2]
```diff
@@ -125,32 +138,50 @@ public Type getType(String typeId)
...
-        return new Type(typeDefinition, dictionaryService, properties);
+        try
+        {
+            Type type = new Type(typeDefinition, dictionaryService);
+            constructFromFilters(type, typeDefinition, includes);
+            return type;
+        }
+        catch (Exception e)
+        {
+            logger.warn("Failed to parse Type" + typeDefinition.getName());
```
Comment (gavincornwell) [inline remote-api/src/main/java/org/alfresco/rest/api/impl/TypesImpl.java:154]: Same comment as above

File: remote-api/src/main/java/org/alfresco/rest/api/model/AssociationSource.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+
+package org.alfresco.rest.api.model;
+
+import java.util.Objects;
+
+public class AssociationSource {
+  private String role = null;
```
Comment (gavincornwell) [inline remote-api/src/main/java/org/alfresco/rest/api/model/AssociationSource.java:32]: Formatting looks different in this file than the others.

**Review by gavincornwell — CHANGES_REQUESTED** [review]

File: remote-api/src/main/java/org/alfresco/rest/api/impl/AspectsImpl.java [hunk 4]
```diff
@@ -151,9 +148,8 @@ public Aspect convertToAspect(AspectDefinition aspectDefinition, List<String> in
         }
         catch (Exception ex)
         {
-            logger.warn("Failed to parse Aspect " + aspectDefinition.getName().toPrefixString());
+            throw new AlfrescoRuntimeException("Failed to parse Aspect" + aspectDefinition.getName());
```
Comment (gavincornwell) [inline remote-api/src/main/java/org/alfresco/rest/api/impl/AspectsImpl.java:151]: Thanks for changing this but looks like you've lost the space in the error message.

File: remote-api/src/main/java/org/alfresco/rest/api/impl/TypesImpl.java [hunk 5]
```diff
@@ -151,9 +148,8 @@ public Type convertToType(TypeDefinition typeDefinition, List<String> includes)
         }
         catch (Exception e)
         {
-            logger.warn("Failed to parse Type" + typeDefinition.getName());
+            throw new AlfrescoRuntimeException("Failed to parse Type" + typeDefinition.getName());
```
Comment (gavincornwell) [inline remote-api/src/main/java/org/alfresco/rest/api/impl/TypesImpl.java:151]: Thanks for changing this but looks like you've lost the space in the error message.

**Review by gavincornwell — CHANGES_REQUESTED** [review]
Looks good other than the missing spaces in error messages.

**Review by gavincornwell — APPROVED** [review]
LGTM

**Review by eromano — APPROVED** [review]

**Pull request merged**